### PR TITLE
Merging dev to main

### PR DESCRIPTION
**Pull request by khawkins**
As we are approaching the next release—and tellingly doc updates in https://github.com/salesforce/eslint-plugin-lwc-graph-analyzer/pull/9 are written against the next release of Komaci—it's time to start working predominantly against that code base. This PR merges `dev` into `main`, accordingly. Only change on my part is rev'ing the package version.

### CODE REVIEW

File: test/lib/rules/artifacts-combined-files/unresolvedIteratorGetter/no-composition-on-unanalyzable-getter-property.js [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 'use strict';
 
+//no-composition-on-unanalyzable-getter-property
```
Comment (maliroteh-sf) [inline test/lib/rules/artifacts-combined-files/unresolvedIteratorGetter/no-composition-on-unanalyzable-getter-property.js:10]: do we want to keep these?

**Review by maliroteh-sf — APPROVED** [review]
LGTM, just left a question for you. Also the file `test/lib/rules/artifacts-combined-files/unresolvableChildComponent/primingDiagnostic.json` is empty. Do we still want to include it in the PR?

File: test/lib/rules/artifacts-combined-files/unresolvedIteratorGetter/no-composition-on-unanalyzable-getter-property.js (hunk 1 above)
Comment (sfdctaka) [reply test/lib/rules/artifacts-combined-files/unresolvedIteratorGetter/no-composition-on-unanalyzable-getter-property.js:10]: This is my oversight on a commit that I made earlier. Must have happened when I was porting tests. Tests do not get packaged and it is not get shipped so we can fix this later too.

**Comment by sfdctaka** [comment]
Empty `primingDiagnostic.json` is another file that got ported from Komaci test. We can remove it.

**Review by sfdctaka — APPROVED** [review]
@khawkins Please let me know if you want me to make a commit to your fork to address things that Meisam found.

**Comment by khawkins** [comment]
> @khawkins Please let me know if you want me to make a commit to your fork to address things that Meisam found.

@sfdctaka @maliroteh-sf I'm totally okay with all of those suggestions. I would prefer they come in the form of a follow-on PR though, just so that this one can be more or less cleanly limited to the branch promotion. Maintenance-wise, it can sometimes be hard to come back to those additional updates to otherwise large-but-routine PR changes.

**Comment by khawkins** [comment]
Another thing we need to do is actually fix the formatting and linting errors that exist in the repo. That was more follow-on work I wanted to do after we merged.

**Pull request merged**